### PR TITLE
feat(ci/cd): Changed lintstaged to prettier-check, also added format and format:fix script, and also added check.yml for prettier & eslint github action

### PR DESCRIPTION
**Pull request by OwenTruong**
I have changed lintstaged to prettier-check so it checks instead of fix/format when someone runs git commit. Also added format and format:fix script in package.json. And also added a check.yml in `.github/workflows/check.yml` for running prettier and eslint check when someone makes a PR request.

### CODE REVIEW

**Comment by ericzty** [comment]
#15

File: .github/workflows/check.yml
Comment (ericzty) [inline .github/workflows/check.yml:1]: Could you split the prettier and lint into two separate test jobs?
let’s also disable fail-fast in the strat, and make them run concurrently. 
Could you also rename it from test to something more descriptive?

**Review by ericzty — CHANGES_REQUESTED** [review]
Nice work, please see comments.

File: .github/workflows/check.yml
Comment (ericzty) [inline .github/workflows/check.yml:1]: we only want the test on ubuntu-latest.

**Review by ericzty — CHANGES_REQUESTED** [review]

File: .github/workflows/check.yml
Comment (ericzty) [reply .github/workflows/check.yml:1]: rest looks good

Comment (OwenTruong) [reply .github/workflows/check.yml:1]: Thanks for the feedback! I am new to this so I appreciate the comments

**Comment by OwenTruong** [comment]
Squashed all 4 commits into one

**Review by miguel-merlin — APPROVED** [review]

**Pull request merged**